diff -dNur a/data/meson.build b/data/meson.build
--- a/data/meson.build	2023-08-22 19:43:58.000000000 +0200
+++ b/data/meson.build	2023-08-23 15:38:27.306286696 +0200
@@ -17,7 +17,7 @@
        output: 'io.missioncenter.MissionCenter.appdata.xml',
        po_dir: '../po',
       install: true,
-  install_dir: join_paths(get_option('datadir'), 'appdata')
+  install_dir: join_paths(get_option('datadir'), 'metainfo')
 )
 
 appstream_util = find_program('appstream-util', required: false)
