diff -dNur a/meson.build b/meson.build
--- a/meson.build	2020-01-02 12:08:47.000000000 +0100
+++ b/meson.build	2020-01-05 18:51:57.029831519 +0100
@@ -92,7 +92,7 @@
         dependency('json-glib-1.0'),
         dependency('gdk-pixbuf-2.0'),
         dependency('libwnck-3.0'),
-        dependency('gtksourceview-3.0'),
+        dependency('gtksourceview-4'),
         m_dep
     ],
     install : true
