diff -dNur a/meson.build b/meson.build
--- a/meson.build	2022-11-26 16:07:41.000000000 +0100
+++ b/meson.build	2023-01-15 21:49:11.248789797 +0100
@@ -72,17 +72,4 @@
 	install : true
 )
 
-ronn = find_program('ronn', required: false)
-if ronn.found()
-	ipcalc_1 = custom_target(
-		'ipcalc.1',
-		output : 'ipcalc.1',
-		input : 'ipcalc.1.md',
-		command : [ronn, '--pipe', '-r', '@INPUT@'],
-		capture: true,
-		install_dir: join_paths(get_option('mandir'), 'man1'),
-		install: true
-	)
-endif
-
 subdir('tests')
