diff -dNur a/etc/qlcplus-fixtureeditor.desktop b/etc/qlcplus-fixtureeditor.desktop
--- a/etc/qlcplus-fixtureeditor.desktop	2015-07-07 12:08:51.000000000 +0000
+++ b/etc/qlcplus-fixtureeditor.desktop	2015-08-28 15:47:19.019183076 +0000
@@ -9,4 +9,4 @@
 Exec=qlcplus-fixtureeditor --open %f
 Icon=qlcplus-fixtureeditor
 MimeType=application/x-qlc-fixture;
-Categories=Lighting;Art;Qt;AudioVideo;
+Categories=X-Lighting;X-Art;Qt;AudioVideo;
diff -dNur a/etc/qlcplus.desktop b/etc/qlcplus.desktop
--- a/etc/qlcplus.desktop	2015-07-07 12:08:51.000000000 +0000
+++ b/etc/qlcplus.desktop	2015-08-28 15:47:08.955131286 +0000
@@ -1,12 +1,12 @@
-[Desktop Entry]
-Type=Application
-Name=Q Light Controller Plus
-GenericName=Lighting control
-GenericName[de]=Lichtsteuerung
-GenericName[fi]=Valojen ohjaus
-GenericName[fr]=Controlleur d'éclairages
-GenericName[it]=Controllore di luci
-Exec=qlcplus --open %f
-Icon=qlcplus
-MimeType=application/x-qlc-workspace;
-Categories=Lighting;Art;Qt;AudioVideo;
+[Desktop Entry]
+Type=Application
+Name=Q Light Controller Plus
+GenericName=Lighting control
+GenericName[de]=Lichtsteuerung
+GenericName[fi]=Valojen ohjaus
+GenericName[fr]=Controlleur d'éclairages
+GenericName[it]=Controllore di luci
+Exec=qlcplus --open %f
+Icon=qlcplus
+MimeType=application/x-qlc-workspace;
+Categories=X-Lighting;X-Art;Qt;AudioVideo;
