--- a/main/main.cpp	2015-08-28 12:49:44.462588456 +0000
+++ b/main/main.cpp	2015-08-28 12:53:03.871565232 +0000
@@ -40,7 +40,6 @@
 
 #include "virtualconsole.h"
 #include "simpledesk.h"
-#include "webaccess.h"
 #include "app.h"
 #include "doc.h"
 
@@ -68,9 +67,6 @@
     /** If true, adjusts the main window geometry instead of instructing the windowing system to "maximize" */
     bool fullScreenResize = false;
 
-    /** If true, create and run a class to enable a web server for remote controlling */
-    bool enableWebAccess = false;
-
     /** If true, enable a 5% of overscan when in fullscreen mode (Raspberry Only) */
     bool enableOverscan = false;
 
@@ -254,10 +250,6 @@
         {
             QLCArgs::operate = true;
         }
-        else if (arg == "-w" || arg == "--web")
-        {
-            QLCArgs::enableWebAccess = true;
-        }
         else if (arg == "-v" || arg == "--version")
         {
             /* Don't print anything, since version is always
@@ -342,18 +334,5 @@
     if (QLCArgs::kioskMode == true && QLCArgs::closeButtonRect.isValid() == true)
         app.createKioskCloseButton(QLCArgs::closeButtonRect);
 
-    if (QLCArgs::enableWebAccess == true)
-    {
-        WebAccess *webAccess = new WebAccess(app.doc(), VirtualConsole::instance(),
-                                               SimpleDesk::instance());
-
-        QObject::connect(webAccess, SIGNAL(toggleDocMode()),
-                &app, SLOT(slotModeToggle()));
-        QObject::connect(webAccess, SIGNAL(loadProject(QString)),
-                &app, SLOT(slotLoadDocFromMemory(QString)));
-        QObject::connect(webAccess, SIGNAL(storeAutostartProject(QString)),
-                &app, SLOT(slotSaveAutostart(QString)));
-    }
-
     return qapp.exec();
 }
