diff -dNur a/configure.ac b/configure.ac
--- a/configure.ac	2017-01-27 21:25:31.000000000 +0100
+++ b/configure.ac	2017-01-31 22:19:12.673295177 +0100
@@ -81,16 +81,6 @@
 AC_SUBST(CB_LIBS)
 AC_SUBST(CB_VALA_FLAGS)
 
-
-# Check for valac >= 0.34, which binds gtk_popover_popup
-# as well as if that function exists in gtk+
-AM_PROG_VALAC([0.34.0], [
-AC_CHECK_LIB([gtk-3], [gtk_popover_popup],
-             [CB_VALA_FLAGS="$CB_VALA_FLAGS -D GTK322"],
-            )
-])
-
-
 AC_ARG_VAR([GLIB_COMPILE_RESOURCES],[the glib-compile-resources program])
 AC_PATH_PROG([GLIB_COMPILE_RESOURCES],[glib-compile-resources],[])
 if test -z "$GLIB_COMPILE_RESOURCES"; then
