diff -dNur a/search-provider/Makefile.am b/search-provider/Makefile.am
--- a/search-provider/Makefile.am	2017-02-23 08:43:51.000000000 +0100
+++ b/search-provider/Makefile.am	2017-03-27 23:23:16.103449177 +0200
@@ -39,21 +39,17 @@
 service_in_files = org.gnome.Lollypop.SearchProvider.service.in
 
 
-appdatadir = $(pkgdatadir)
-appdata_DATA = lollypop-sp
+libexec_SCRIPTS = lollypop-sp
 
 EXTRA_DIST = \
         org.gnome.Lollypop.SearchProvider.service.in\
         lollypop-sp.in\
-	$(appdata_DATA)\
+	$(libexec_SCRIPTS)\
 	$(searchprovider_DATA) \
 	$(service_DATA)
 
 CLEANFILES = \
-        $(appdata_DATA)\
+        $(libexec_SCRIPTS)\
 	$(service_DATA)
 
-install-data-hook:
-	test -f $(pkgdatadir)/$(appdata_DATA) && chmod +x $(pkgdatadir)/$(appdata_DATA); exit 0
-
 -include $(top_srcdir)/git.mk
diff -dNur a/search-provider/org.gnome.Lollypop.SearchProvider.service.in b/search-provider/org.gnome.Lollypop.SearchProvider.service.in
--- a/search-provider/org.gnome.Lollypop.SearchProvider.service.in	2017-02-23 08:43:51.000000000 +0100
+++ b/search-provider/org.gnome.Lollypop.SearchProvider.service.in	2017-03-27 23:19:03.035317026 +0200
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=org.gnome.Lollypop.SearchProvider
-Exec=@pkgdatadir@/lollypop-sp
+Exec=@libexecdir@/lollypop-sp
