From 6ab0ddec71d0bd5dbebae29855c4dd13b758db4c Mon Sep 17 00:00:00 2001
From: Jente Hidskes <hjdskes@gmail.com>
Date: Sun, 15 Oct 2017 12:31:58 +0200
Subject: [PATCH 19/30] RatbagdLed: fix docstring

---
 piper/ratbagd.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/piper/ratbagd.py b/piper/ratbagd.py
index 9e4000a..4290aa4 100644
--- a/piper/ratbagd.py
+++ b/piper/ratbagd.py
@@ -926,8 +926,7 @@ class RatbagdLed(_RatbagdDBus):
 
     @GObject.Property
     def type(self):
-        """An enum describing this led's type,
-        RatbagdLed.TYPE_LOGO or RatbagdLed.TYPE_SIDE."""
+        """An enum describing this led's type, one of RatbagdLed.TYPE_*."""
         return self._get_dbus_property("Type")
 
     @GObject.Property
-- 
2.16.1

