From 4f560b0d1e13383a075dfe94de106942a2028860 Mon Sep 17 00:00:00 2001
From: Jente Hidskes <hjdskes@gmail.com>
Date: Thu, 14 Sep 2017 16:07:51 +0200
Subject: [PATCH 13/30] ratbagd: sync with libratbag

---
 piper/ratbagd.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/piper/ratbagd.py b/piper/ratbagd.py
index e74039d..4478cf2 100644
--- a/piper/ratbagd.py
+++ b/piper/ratbagd.py
@@ -623,6 +623,7 @@ class RatbagdButton(_RatbagdDBus):
     ACTION_TYPE_SPECIAL = 2
     ACTION_TYPE_MACRO = 4
 
+    ACTION_SPECIAL_INVALID = -1
     ACTION_SPECIAL_UNKNOWN = (1 << 30)
     ACTION_SPECIAL_DOUBLECLICK = (1 << 30) + 1
     ACTION_SPECIAL_WHEEL_LEFT = (1 << 30) + 2
@@ -684,6 +685,7 @@ class RatbagdButton(_RatbagdDBus):
 
     """A table mapping a special function to its human-readable description."""
     SPECIAL_DESCRIPTION = {
+        ACTION_SPECIAL_INVALID: N_("Invalid"),
         ACTION_SPECIAL_UNKNOWN: N_("Unknown"),
         ACTION_SPECIAL_DOUBLECLICK: N_("Doubleclick"),
         ACTION_SPECIAL_WHEEL_LEFT: N_("Wheel Left"),
-- 
2.16.1

