diff -dNur a/configure.ac b/configure.ac
--- a/configure.ac	2017-06-14 14:20:42.000000000 +0200
+++ b/configure.ac	2018-07-25 17:19:37.832581622 +0200
@@ -29,6 +29,11 @@
 PKG_CHECK_MODULES([jose], [jose >= 8])
 PKG_CHECK_MODULES([systemd], [systemd])
 
+AC_ARG_ENABLE([nagiosplugin],
+              AS_HELP_STRING([--disable-nagios], [Disable Nagios plugin]))
+
+AM_CONDITIONAL([NAGIOSPLUGIN], [test "x$enable_nagiosplugin" != "xno"])
+
 AC_ARG_WITH([systemdsystemunitdir],
             [AS_HELP_STRING([--with-systemdsystemunitdir=DIR], [Directory for systemd service files])],
             [],
diff -dNur a/Makefile.am b/Makefile.am
--- a/Makefile.am	2017-06-10 15:29:39.000000000 +0200
+++ b/Makefile.am	2018-07-25 17:21:39.342292389 +0200
@@ -16,12 +16,16 @@
 
 dist_libexec_SCRIPTS = src/tangd-update src/tangd-keygen
 libexec_PROGRAMS = src/tangd
+if NAGIOSPLUGIN
 nagios_PROGRAMS = src/tang
 man1_MANS = doc/tang-nagios.1
+endif
 man8_MANS = doc/tang.8
 
 src_tangd_SOURCES = src/http.c src/http.h src/tangd.c
+if NAGIOSPLUGIN
 src_tang_SOURCES = src/nagios.c
+endif
 
 %: %.in
 	$(AM_V_GEN)mkdir -p "`dirname "$@"`"
@@ -32,12 +36,18 @@
 		$(srcdir)/$@.in > $@
 
 AM_TESTS_ENVIRONMENT = SD_ACTIVATE="@SD_ACTIVATE@" PATH=$(srcdir)/src:$(builddir)/src:$(PATH)
-TESTS = tests/adv tests/rec tests/nagios
+TESTS = tests/adv tests/rec
+if NAGIOSPLUGIN
+TESTS += tests/nagios
+endif
 
 CLEANFILES = $(nodist_systemdsystemunit_DATA)
 EXTRA_DIST = \
     $(foreach unit,$(nodist_systemdsystemunit_DATA),$(unit).in) \
     COPYING \
     $(TESTS) \
-    $(man1_MANS) \
     $(man8_MANS)
+
+if NAGIOSPLUGIN
+EXTRA_DIST += $(man1_MANS)
+endif
