From 17f3ef6bea533b4d6e90419bf0346607c73db353 Mon Sep 17 00:00:00 2001
From: Uwe Klotz <uklotz@mixxx.org>
Date: Mon, 16 Apr 2018 07:23:37 +0200
Subject: [PATCH 1/2] Fix invocation args of pasuspender

---
 res/linux/mixxx.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/res/linux/mixxx.desktop b/res/linux/mixxx.desktop
index 14c967c286..5e1ff650b0 100644
--- a/res/linux/mixxx.desktop
+++ b/res/linux/mixxx.desktop
@@ -8,7 +8,7 @@ GenericName[fr]=Interface numérique pour DJ
 Comment=A digital DJ interface
 Comment[de]=Ein digitales DJ-System
 Comment[fr]=Une interface numérique pour DJ
-Exec=sh -c "pasuspender mixxx || mixxx"
+Exec=sh -c "pasuspender -- mixxx || mixxx"
 Terminal=false
 Icon=mixxx-icon
 Type=Application

From 271432773dae5e8d860106ba3dbe5b1757797471 Mon Sep 17 00:00:00 2001
From: Uwe Klotz <uklotz@mixxx.org>
Date: Mon, 16 Apr 2018 12:03:38 +0200
Subject: [PATCH 2/2] Fix remaining invocations with of pasuspender

---
 build/debian/menu          | 2 +-
 build/debian/mixxx.desktop | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/debian/menu b/build/debian/menu
index c945aa3d19..b1ff680748 100644
--- a/build/debian/menu
+++ b/build/debian/menu
@@ -2,6 +2,6 @@
   needs="X11" \
   section="Apps/Sound" \
   title="Mixxx" \
-  command="/usr/bin/pasuspender /usr/bin/mixxx" \
+  command="/usr/bin/pasuspender -- /usr/bin/mixxx" \
   icon="/usr/share/pixmaps/mixxx.xpm" \
   icon16x16="/usr/share/pixmaps/mixxx-16.xpm"
diff --git a/build/debian/mixxx.desktop b/build/debian/mixxx.desktop
index 45323b5c15..947ef82a20 100644
--- a/build/debian/mixxx.desktop
+++ b/build/debian/mixxx.desktop
@@ -3,7 +3,7 @@ Version=1.0
 Name=Mixxx
 GenericName=Digital DJ interface
 Comment=A digital DJ interface
-Exec=pasuspender mixxx
+Exec=pasuspender -- mixxx
 Terminal=false
 Icon=mixxx-icon
 Type=Application
