diff -dNur a/src/waveform/vsyncthread.cpp b/src/waveform/vsyncthread.cpp
--- a/src/waveform/vsyncthread.cpp	2015-12-17 02:44:04.000000000 +0100
+++ b/src/waveform/vsyncthread.cpp	2015-12-19 17:29:19.449521628 +0100
@@ -1,4 +1,5 @@
 #include <QThread>
+#include <QGLWidget>
 #include <QGLFormat>
 #include <QTime>
 #include <QtDebug>
@@ -131,12 +132,8 @@
     glw->swapBuffers();
 #else
 #if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
-#ifdef QT_OPENGL_ES_2
-    glw->swapBuffers();
-#else
     const QX11Info *xinfo = qt_x11Info(glw);
     glXSwapBuffers(xinfo->display(), glw->winId());
-#endif
 #else
     glw->swapBuffers();
 #endif // QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
diff -dNur a/src/waveform/vsyncthread.h b/src/waveform/vsyncthread.h
--- a/src/waveform/vsyncthread.h	2015-12-17 02:44:04.000000000 +0100
+++ b/src/waveform/vsyncthread.h	2015-12-19 17:29:19.449521628 +0100
@@ -5,7 +5,6 @@
 #include <QThread>
 #include <QSemaphore>
 #include <QPair>
-#include <QGLWidget>
 
 #if defined(__APPLE__)
 
@@ -13,19 +12,26 @@
 
 #else
 #if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
-#ifndef QT_OPENGL_ES_2
     #include <qx11info_x11.h>
+#endif // QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
+#endif
+
+#include "util/performancetimer.h"
+
+
+#if defined(__APPLE__)
+
+#elif defined(__WINDOWS__)
+
+#else
     #include <GL/glx.h>
-    //#include "GL/glxext.h"
+    #include "GL/glxext.h"
     // clean up after Xlib.h, which #defines values that conflict with QT.
     #undef Bool
     #undef Unsorted
-#endif // QT_OPENGL_ES_2
-#endif // QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
 #endif
 
-#include "util/performancetimer.h"
-
+class QGLWidget;
 class GuiTick;
 class MixxxMainWindow;
 
