diff -dNur a/src/soundsourceffmpeg.cpp b/src/soundsourceffmpeg.cpp
--- a/src/soundsourceffmpeg.cpp	2015-09-18 11:48:36.423194604 +0000
+++ b/src/soundsourceffmpeg.cpp	2015-09-18 11:56:15.933501047 +0000
@@ -509,7 +509,7 @@
 
     m_iCurrentMixxTs = filepos;
 
-    m_bIsSeeked = TRUE;
+    m_bIsSeeked = true;
 
     return filepos;
 }
@@ -521,7 +521,7 @@
         // Make sure we allways start at begining and cache have some
         // material that we can consume.
         seek(0);
-        m_bIsSeeked = FALSE;
+        m_bIsSeeked = false;
     }
 
     getBytesFromCache((char *)destination, m_iCurrentMixxTs, size);
@@ -530,11 +530,11 @@
     //  As this is also Hack
     // If we don't seek like we don't on analyzer.. keep
     // place in mind..
-    if (m_bIsSeeked == FALSE) {
+    if (m_bIsSeeked == false) {
         m_iCurrentMixxTs += size;
     }
 
-    m_bIsSeeked = FALSE;
+    m_bIsSeeked = false;
     return size;
 
 }
