diff -dNur a/src/effects/backends/effectsbackendmanager.cpp b/src/effects/backends/effectsbackendmanager.cpp
--- a/src/effects/backends/effectsbackendmanager.cpp	2024-02-16 00:23:29.000000000 +0100
+++ b/src/effects/backends/effectsbackendmanager.cpp	2024-03-22 21:08:51.652593596 +0100
@@ -2,6 +2,7 @@
 
 #include "control/controlobject.h"
 #include "effects/backends/builtin/builtinbackend.h"
+#include "effects/backends/effectmanifest.h"
 #include "effects/backends/effectprocessor.h"
 #ifdef __LILV__
 #include "effects/backends/lv2/lv2backend.h"
