From ae10b48dff54a07ea638421554cab207b8bb1ced Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 2 Dec 2014 18:28:39 +0100
Subject: bookmarks: Only emit "item removed" if actually removed


diff --git a/src/bookmarks/grl-bookmarks.c b/src/bookmarks/grl-bookmarks.c
index 46e9c30..c7f0109 100644
--- a/src/bookmarks/grl-bookmarks.c
+++ b/src/bookmarks/grl-bookmarks.c
@@ -631,7 +631,7 @@ remove_bookmark (GrlBookmarksSource *bookmarks_source,
 
   g_object_unref (resource);
 
-  if (bookmarks_source->priv->notify_changes) {
+  if (*error != NULL && bookmarks_source->priv->notify_changes) {
     /* We can improve accuracy computing the parent container of removed
        element */
     grl_source_notify_change (GRL_SOURCE (bookmarks_source),
-- 
cgit v0.10.2

