diff -dNur a/src/ttf.c b/src/ttf.c
--- a/src/ttf.c	2011-06-12 23:43:39.000000000 +0200
+++ b/src/ttf.c	2014-07-24 16:54:19.322207452 +0200
@@ -25,8 +25,8 @@
 #include <assert.h>
 
 #include <ft2build.h>
-#include <freetype/ftoutln.h>
-#include <freetype/ttnameid.h>
+#include FT_OUTLINE_H
+#include FT_TRUETYPE_IDS_H
 
 #include "common.h"
 #include "render.h"
diff -dNur a/src/ttf.h b/src/ttf.h
--- a/src/ttf.h	2011-06-12 23:43:39.000000000 +0200
+++ b/src/ttf.h	2014-07-24 16:54:30.426027477 +0200
@@ -1,8 +1,8 @@
 #ifndef _TTF_H
 #define _TTF_H
 #include <ft2build.h>
-#include <freetype/ftoutln.h>
-#include <freetype/ttnameid.h>
+#include FT_OUTLINE_H
+#include FT_TRUETYPE_IDS_H
 
 #define CACHED_METRICS  0x10
 #define CACHED_BITMAP   0x01
