diff -dNur a/configure.ac b/configure.ac
--- a/configure.ac	2018-03-21 19:16:33.000000000 +0100
+++ b/configure.ac	2018-07-26 11:00:36.263883657 +0200
@@ -17,7 +17,6 @@
 PKG_CHECK_MODULES([udisks2], [udisks2])
 PKG_CHECK_MODULES([jose], [jose >= 8])
 PKG_CHECK_MODULES([systemd], [systemd])
-PKG_CHECK_MODULES([dracut], [dracut])
 PKG_CHECK_MODULES([audit], [audit >= 2.7.8])
 
 AC_CHECK_PROG([PWMAKE], [pwmake], [yes])
